### ISSUE TITLE
Ignore abstract functions for unified-signatures rule

### ISSUE THREAD

**Issue by petermorlion** [palantir/tslint#4176]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
abstract class Foo {
    public abstract Bar(): void;
    public abstract Bar(data: any): void;
}
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {},
    "rulesDirectory": []
}
```

#### Actual behavior

I'm getting the error `These overloads can be combined into one signature with an optional parameter.`

I could change it to:

```ts
abstract class Foo {
    public abstract Bar(data: any = null): void;
}
```

But of course this isn't ok for typescript because parameter initializers only work for functions that have a body.

#### Expected behavior

Ignore abstract function for the rule [unified-signatures ](https://palantir.github.io/tslint/rules/unified-signatures/).

**Comment by JoshuaKGoldberg** [comment]
This would be good as an option on the rule. Related: #2433.

**Comment by petermorlion** [comment]
Hm, would it make sense to want to keep this option off? I would expect tslint to always ignore abstract functions for this rule. But I may be overlooking some cases.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
